src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -324,6 +324,19 @@ def run_single(
                 provenance_repo_url, provenance_commit_digest = extract_repo_and_commit_from_provenance(
                     provenance_payload
                 )
+                # Check the provenance repo commit match the input repo commit
```

Comment:
Just some general questions. I can see that this PR only handles the case where repo url, commit hash are available as CLI options, and obtainable within the input provenance:
- Are we planning to perform the validation when the repo url or commit hash are obtainable from the input PURL string too?
- Are we planning to handle the case where repo url is provide but not commit hash ? 
(they could be out of scope, and could be handled in a different Pull request)
:thinking: